src/Terminology.hs

```diff
@@ -88,6 +91,29 @@ instance Showable Call where
     toHtml Redouble  = "Rdb"
     toHtml (Bid l s) = show l ++ toHtml s
 
+-- Calls might have alerted explanations. We always show an alert from the
+-- opponents, but only show alerts from our side in the solutions.
+data CallExplanation = Unalerted | OppsAlerted String | WeAlerted String
+
+instance Showable CallExplanation where
+    toLatex Unalerted = ""
+    toLatex (OppsAlerted e) = "\\footnote{" ++ e ++ "}"
+    toLatex (WeAlerted e) =
+         "\\ifdefined\\showsolutions" ++ toLatex (OppsAlerted e) ++ "\\fi"
```

Comment:
Make `\ourAlert` and `\oppsAlert` macros, and use those instead of plumbing the `\ifdefined\showsolutions` all the way to the Haskell side. Abstract out the implementation!